### ISSUE TITLE
Forwarded Fixed jsonldContext not reapplied when querying type

### ISSUE THREAD

**Issue by jason-fox** [ScorpioBroker/ScorpioBroker#583]
Scorpio 4.1.15. I have a registration to a context source as shown:

```console
curl -L 'http://localhost:1026/ngsi-ld/v1/csourceRegistrations/' \
-H 'Link: <http://context/user-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json' \
-H 'Content-Type: application/json' \
-d '{
    "type": "ContextSourceRegistration",
    "information": [
        {"entities": [{"type": "Building"}]}
    ],
    "contextSourceInfo": [
        {
            "key": "jsonldContext",
            "value": "http://context/fixed-context.jsonld"
        }
    ],
    "mode": "inclusive",
    "operations": ["federationOps"],
    "endpoint": "http://adapter:3000"
}'
```

It is important to note that the `type`: `https://uri.fiware.org/ns/dataModels#Building` is defined differently within the `user-context.jsonld` and `fixed-context.jsonld` files:

### User-Context (Scorpio)

```json
{
   "Building": "https://uri.fiware.org/ns/dataModels#Building"
}
```

### Fixed-Context (Registrant)

```json
{
   "Store": "https://uri.fiware.org/ns/dataModels#Building"
}
```

### Get Entity

When I receive a specific entity which is present on the Registrant:

```console
curl -L 'http://localhost:1026/ngsi-ld/v1/entities/urn:ngsi-ld:Store:001?pick=id%2Ctype' \
-H 'Link: <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json'
```

```json
{
    "id": "urn:ngsi-ld:Store:001",
    "type": "Building"
}
```

The entity is found on the Registrant as `type:Store`, the fixed Context is applied `"Store": "https://uri.fiware.org/ns/dataModels#Building"` and compacted down to `Building`  ✅ 

### Query Entity by `id`

```console
curl -L 'http://localhost:1026/ngsi-ld/v1/entities/?id=urn%3Angsi-ld%3AStore%3A001&pick=id%2Ctype' \
-H 'Link: <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json'
```

The entity `id` is found on the Registrant as `type:Store`, the fixed Context is applied `"Store": "https://uri.fiware.org/ns/dataModels#Building"` and compacted down to `Building` ✅ 

### Query Entity by `type`

```console
curl -L 'http://localhost:1026/ngsi-ld/v1/entities/?type=Building&pick=id%2Ctype' \
-H 'Link: <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json'
```

The entity `type:Building` is found on the Registration, but the fixed Context is not being applied to the query string - hence the request  is not being JSON-LD expanded/compacted  from `Building`-> `https://uri.fiware.org/ns/dataModels#Building`   -> `Store`. The registrant is not receiving a request for  `/entities?type=Building&pick=id,type` and the entity is not found  ❌

**Comment by jason-fox** [comment]
From the 1.8.1 [ETSI Specification](https://www.etsi.org/deliver/etsi_gs/CIM/001_099/009/01.08.01_60/gs_cim009v010801p.pdf)

> **4.3.6.6 Additional pre- and post-processing of extra information when contacting Context Source**
>
>The following key-values have a specific well-defined meaning when defined as elements within the optional array
>contextSourceInfo of the CSourceRegistration.
> - If the key `"jsonldContext"` is defined, the value shall correspond to a URL reference as defined by the
JSON-LD specification [2], section 3.1.
> - The Context Broker shall apply a compaction operation as defined by the JSON-LD specification [2], section
> 4.1.5 over both payload and query parameters using the JSON-LD Context supplied in the value of the
> "jsonldContext" key-value pair, prior to distributing the request to the context source endpoint and
> forwarding with this JSON-LD context using an appropriate binding.

**Comment by ScorpioBroker** [comment]
Fixed in repo. Next release will be tomorrow or the day after

**Comment by ScorpioBroker** [comment]
Fixed in 5.0.0
